master/internal/rm/kubernetesrm/spec_test.go

```diff
@@ -219,3 +220,42 @@ func TestAllPrintableCharactersInEnv(t *testing.T) {
 	require.Contains(t, actual, k8sV1.EnvVar{Name: "test2", Value: ""})
 	require.Contains(t, actual, k8sV1.EnvVar{Name: "func", Value: "f(x)=x"})
 }
+
+func TestDeterminedLabels(t *testing.T) {
+	// fill out task spec
+	taskSpec := tasks.TaskSpec{
+		Owner:       createUser(),
+		Workspace:   "test-workspace",
+		TaskType:    model.TaskTypeCommand,
+		TaskID:      model.NewTaskID().String(),
+		ContainerID: "container-id",
+	}
+
+	p := pod{
+		req: &sproto.AllocateRequest{
+			ResourcePool: "test-rp",
+		},
+		submissionInfo: &podSubmissionInfo{
+			taskSpec: taskSpec,
+		},
+	}
+
+	// define expectations
+	expectedLabels := map[string]string{
+		userLabel:         taskSpec.Owner.Username,
+		workspaceLabel:    taskSpec.Workspace,
+		resourcePoolLabel: p.req.ResourcePool,
+		taskTypeLabel:     string(taskSpec.TaskType),
+		taskIDLabel:       taskSpec.TaskID,
+		containerIDLabel:  taskSpec.ContainerID,
+	}
+
+	spec := p.configurePodSpec(make([]k8sV1.Volume, 1), k8sV1.Container{},
+		k8sV1.Container{}, make([]k8sV1.Container, 1), &k8sV1.Pod{}, "scheduler")
+
+	// confirm pod spec has required labels
+	require.NotNil(t, spec)
+	for expectedKey, expectedValue := range expectedLabels {
+		require.Equal(t, spec.ObjectMeta.Labels[expectedKey], expectedValue)
+	}
```

Comment:
Except it's ok if there's extra labels; the goal is to ensure that the required labels make it to the pod.

Reply:
Why is that okay? I think the name ```TestDeterminedLabels``` should make sure we have the correct labels present on the pod? 

To me if we added the label ```"test": "true"``` to the pod that would be a bug since we aren't expecting. I would want the test to catch that case